models/faculty.js

```diff
@@ -0,0 +1,95 @@
+const { connector } = require('./databaseUtil');
+
+const facultySchema = new mongoose.Schema({
+  name: {
+    type: String,
+    required: true,
+  },
+  department: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Department',
+    required: true,
+  },
+  empType: {
+    type: String,
+    required: true,
+  },
+  emdUID: {
+    type: String,
+    required: true,
+  },
+  preferredSubjects: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Subject',
+    required: true,
+  },

```

Comment:
I'm sorry if i wasn't clear before, but since Subjects here can be more than one, we need this to be an array of objects, so simply put square brackets around the curly brackets, and that'll be all.